### DISCUSSION TITLE
I see that in client.c in the C API, select is used for socket listening. Does this mean the C API does not support Tokio?

### DISCUSSION THREAD

**Discussion by spongehah** [hyperium/hyper#3710] [Q&A]

**Comment by seanmonstar** [comment]
The C API works similar to the Rust API: you bring your own runtime. The example shows how to use it with select, but you can also use poll, epoll, kqueue, whatever you like.

hyper even in Rust does not bundle Tokio with it. There's some helper types in `hyper_util::rt`.
